scripts/github/changesets-publish.ts

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable @typescript-eslint/no-unsafe-call, @typescript-eslint/no-unsafe-return,@typescript-eslint/no-unsafe-assignment, unicorn/prefer-string-raw */
+
+import { globSync } from 'glob';
+import { execSync } from 'node:child_process';
+import fs from 'node:fs';
+import path from 'node:path';
+import { fileURLToPath } from 'node:url';
+
+const __filename = fileURLToPath(import.meta.url);
+const __dirname = path.dirname(__filename);
+
+function run(cmd: string, options = {}): any {
+	try {
+		return execSync(cmd, {
+			stdio: 'pipe',
+			encoding: 'utf8',
+			...options
+		}).trim();
+	} catch (error: any) {
+		if (error.stdout) {
+			return error.stdout.toString();
+		}
+
+		throw error;
+	}
+}
+
+/**
+ * Get the current version from package.json - we can use foundations here as it's always in sync with the other packages
+ */
+function getVersion(): string {
+	const pkgPath = path.resolve(
+		__dirname,
+		'../../packages/foundations/package.json'
+	);
+	const pkg = JSON.parse(fs.readFileSync(pkgPath, 'utf8'));
+	return pkg.version;
+}
+
+function tagExists(tag: string): boolean {
+	const tags = run(`git tag --list "${tag}"`);
+	return tags.split('\n').includes(tag);
+}
+
+/**
+ * Find all CHANGELOG.md files in the repo, excluding node_modules, using glob.
+ */
+function findChangelogFiles(repoRoot: string): string[] {
+	return globSync('**/CHANGELOG.md', {
+		cwd: repoRoot,
+		ignore: ['**/node_modules/**'],
```

Comment:
```suggestion
		cwd: repoRoot + 'packages/',
```

as we're releasing packages only out of the `packages` folder, couldn't we define this one as the root to search for `CHANGELOG.md` files?